### PR TITLE
Add ability to disable dates in the future and/or past

### PR DESCRIPTION
**Pull request by hahahana**
Live example [here](http://cdn.rawgit.com/hahahana/react-datepicker/23f3362c1e3b7be530d2891fe4238be13a1c5469/example.html).

The example is getting a bit gnarly, what do you guys think? Maybe separate pages for different functionality with some documentation on the page too?

---
Should close https://github.com/Hacker0x01/react-datepicker/issues/69

### CODE REVIEW

File: src/day.js [hunk 1]
```diff
@@ -4,13 +4,14 @@ var Day = React.createClass({
...
       'datepicker__day': true,
+      'datepicker__day--disabled': this.props.disabled,
       'datepicker__day--selected': this.props.day.sameDay(this.props.selected),
       'datepicker__day--this-month': this.props.day.sameMonth(this.props.date),
       'datepicker__day--today': this.props.day.sameDay(moment())
     });
 
     return (
-      <div className={classes} onClick={this.props.onClick}>
+      <div className={classes} onClick={this.props.disabled ? null : this.props.onClick}>
```
Comment (RSO) [inline src/day.js]: I think that I'd rather have this logic inside of it's own method:

```
handleClick: function(event) {
  if (this.props.disabled) return;

  this.props.onClick(event)
}
```

**Comment by RSO** [comment]
Yeah, I agree about the example. I was hoping to trick @lbekkema or @jenskanis in designing a Github page for this component, with documentation etc.

Great job on the PR, I'll let @martijnrusschen decide if it's alright with him, but it looks good to go! Awesomesaus!

**Comment by martijnrusschen** [comment]
Thanks, looks great! I agree about the example, at first that was a quick hack to show the datepicker, but with these different configurations it becomes really ugly. I'll try to make something better this weekend.

**Comment by jenskanis** [comment]
@RSO we can generate a Github page with minimal effort, but we need to have sufficient permissions of the repo. Can you give us that?

**Pull request merged**